hack/update-codegen-dockerized.sh

```diff
@@ -122,96 +134,146 @@ if [[ "$#" -eq 1 && $1 == "mockgen" ]]; then
+  $GOPATH/bin/go-to-protobuf \
+    --proto-import /tmp/includes \
+    --packages "${PACKAGES}" \
+    --go-header-file hack/boilerplate/license_header.go.txt
+  rm -rf /tmp/includes
+}
 
-# Download vendored modules to the vendor directory so it's easier to
-# specify the search path of required protobuf files.
-go mod vendor
-# In Go 1.14, vendoring changed (see release notes at
-# https://golang.org/doc/go1.14), and the presence of a go.mod file specifying
-# go 1.14 or higher causes the go command to default to -mod=vendor when a
-# top-level vendor directory is present in the module. This causes the
-# go-to-protobuf command below to complain about missing packages under vendor/,
-# which were not downloaded by "go mod vendor". We can workaround this easily by
-# renaming the vendor directory.
-mv vendor /tmp/includes
-PACKAGES="${ANTREA_PKG}/pkg/apis/stats/v1alpha1=${ANTREA_PROTO_PKG}.pkg.apis.stats.v1alpha1,\
-${ANTREA_PKG}/pkg/apis/controlplane/v1beta2=${ANTREA_PROTO_PKG}.pkg.apis.controlplane.v1beta2"
-$GOPATH/bin/go-to-protobuf \
-  --proto-import /tmp/includes \
-  --packages "${PACKAGES}" \
-  --go-header-file hack/boilerplate/license_header.go.txt
-rm -rf /tmp/includes
+function generate_multicluster_codes {
+  # Generate clientset and apis code with K8s codegen tools.
+  $GOPATH/bin/client-gen \
+    --clientset-name versioned \
+    --input-base "${ANTREA_PKG}/multicluster/apis" \
+    --input "multicluster/v1alpha1" \
+    --input "multicluster/v1alpha2" \
+    --output-package "${ANTREA_PKG}/multicluster/pkg/client/clientset" \
+    --go-header-file hack/boilerplate/license_header.go.txt
+  
+  # Generate listers with K8s codegen tools.
+  $GOPATH/bin/lister-gen \
+    --input-dirs "${ANTREA_PKG}/multicluster/apis/multicluster/v1alpha1" \
+    --input-dirs "${ANTREA_PKG}/multicluster/apis/multicluster/v1alpha2" \
+    --output-package "${ANTREA_PKG}/multicluster/pkg/client/listers" \
+    --go-header-file hack/boilerplate/license_header.go.txt
+  
+  # Generate informers with K8s codegen tools.
+  $GOPATH/bin/informer-gen \
+    --input-dirs "${ANTREA_PKG}/multicluster/apis/multicluster/v1alpha1" \
+    --input-dirs "${ANTREA_PKG}/multicluster/apis/multicluster/v1alpha2" \
+    --versioned-clientset-package "${ANTREA_PKG}/multicluster/pkg/client/clientset/versioned" \
+    --listers-package "${ANTREA_PKG}/multicluster/pkg/client/listers" \
+    --output-package "${ANTREA_PKG}/multicluster/pkg/client/informers" \
+    --go-header-file hack/boilerplate/license_header.go.txt
+
+  $GOPATH/bin/controller-gen object:headerFile="hack/boilerplate/license_header.go.txt",year=$(date "+%Y") paths="./multicluster/..."
+}
+
+if [[ "$#" -eq 1 && $1 == "multicluster" ]]; then
+  generate_multicluster_codes
+  reset_year_change
+  exit 0
+fi
+
+echo "===== Generating default client codes ====="
+generate_default_codes
+
+echo "===== Generating multicluter client codes ====="
+generate_multicluster_codes
```

Comment:
So by default we generate code for both antrea and multicluster? Why make this behavior change?

Reply:
There is a workflow which will run `make codegen`, I am thinking it's better to let it regenerate MC's client code as well. But there is another approach, I can refine the workflow instead of here.

Reply:
I feel better to keep the current behavior, as Make in the root will not build any other thing in multicluster.

Reply:
Done